packages/next/next-server/server/config-shared.ts

```diff
@@ -104,7 +105,6 @@ export const defaultConfig: NextConfig = {
     plugins: false,
     profiling: false,
     sprFlushToDisk: true,
-    reactMode: (process.env.NEXT_PRIVATE_REACT_MODE as any) || 'legacy',
```

Comment:
We'll want to throw an error when `reactMode` is set to highlight the changes to users that were using it.

Reply:
Took it a bit further and defaulted the new value too.